rundetection/rules/rule.py

```diff
@@ -2,20 +2,20 @@
 Module containing the abstract base Rule class and MissingRuleError
 """
 from abc import ABC, abstractmethod
-from typing import Generic, TypeVar, List
+from typing import Generic, TypeVar
 
 from rundetection.ingest import DetectedRun
 
-T_co = TypeVar("T_co", str, bool, int, float, None, List[str], covariant=True)
+T = TypeVar("T")
```

Comment:
Mypy seems to have an issue stemming from bools being ints in python and the way covariant generics are implemented